test-tests/shared/src/test/scala/zio/test/SmartAssertionSpec.scala

```diff
@@ -71,6 +71,18 @@ object SmartAssertionSpec extends ZIOBaseSpec {
       val myString = "something"
       assertTrue(myString.contains("aoseunoth") && myString == "coool")
     } @@ failing,
+    test("or succeeds if lhs side succeeds") {
+      val myOption = Option.empty[Int]
+      assertTrue(myOption.isEmpty || myOption.get == 1)
+    },
+    test("or succeeds if rhs side succeeds") {
+      val myOption = Option.empty[Int]
+      assertTrue(myOption.get == 1 || myOption.isEmpty)
+    },
```

Comment:
I'm not sure we want this semantics for rhs, though.
For example, such boolean expression will fail in plain Scala `option.get == 1 || option.isEmpty`.